etna/records/tests/test_templatetags.py

```diff
@@ -267,3 +623,59 @@ def test_non_reference_number_url(self):
                     record_url(source.repository, use_non_reference_number_url=True),
                     expected_result,
                 )
+
+    def test_is_page_current_item_in_hierarchy(self):
+        current_record = self.tna_long_hierarchy_record_instance
+        with self.subTest(current_record):
+            # We pass in the record and the current position in the hierearchy
+            # which we have looped through. This function then checks if the
+            # current record is the same as the record in the hierarchy at the
+            # current position
+            for record in self.tna_long_hierarchy_record_instance.hierarchy:
+                if record.iaid == current_record.iaid:
+                    self.assertEqual(
+                        is_page_current_item_in_hierarchy(current_record, record),
+                        True,
+                    )
+                else:
+                    self.assertEqual(
+                        is_page_current_item_in_hierarchy(current_record, record),
+                        False,
+                    )
```

Comment:
```suggestion
    def test_is_page_current_item_in_hierarchy(self):
page = self.tna_long_hierarchy_record_instance
        hierarchy = page.hierarchy
        for name, level_item, expected_result in (("item_level_1", hierarchy[0], False),
                                                  ("item_level_3", hierarchy[1], False),
                                                  ("item_level_6", hierarchy[2], False),
                                                  ("item_level_7", hierarchy[3], True)
                                                  ):
            with self.subTest(name):
                self.assertEqual(is_page_current_item_in_hierarchy(page, level_item),expected_result)
```
@jamesbiggs , really good to see a test case for this.
1) should avoid conditions like these `if record.iaid == current_record.iaid:` in a test case, we try on only check with the data provided 
2) ` is_page_current_item_in_hierarchy(current_record, record)`  its actually the page, item which passes through see `def is_page_current_item_in_hierarchy(page: Record, hierarchy_item: Record):`

See my suggested change for the above.

Reply:
Thanks Allan, I was trying to think of a way to do this for ages 😅 Thanks for the suggestion! 😃